source/processes/electromagnetic/lowenergy/src/G4PenelopeComptonModel.cc

```diff
@@ -390,10 +390,12 @@ void G4PenelopeComptonModel::SampleSecondaries(std::vector<G4DynamicParticle*>*
       G4double cdt1 = 0.;
       do
 	{
+    do{
 	  if ((G4UniformRand()*a2) < a1)
 	    tau = std::pow(taumin,G4UniformRand());
 	  else
 	    tau = std::sqrt(1.0+G4UniformRand()*(taumin*taumin-1.0));
+    } while (tau==1.0);
```

Comment:
@snipdome Thanks again, just trying to understand the details. The do-while{} at l. 393-398 is meant for the cases in which tau is stuck to 1 because of FP? What's the energy that you are considering for the gamma-rays? Is the issue affecting both branches "energy>5 MeV" and "energy < 5 MeV", right?

Reply:
I tested for energy < 5MeV. As the lines 395 and 397 are the same for > 5 MeV, then I assume the code will have the same problem for these energy range as well.
Regarding the while at 398 (added by me), it has been added to prevent that tau is 1.0. If there would not be the while, then the code goes on and it finally gets stuck after many lines and many operations, at the loop in 456.

Reply:
Thanks a lot for the diagnostics, I am going to have a closer look! Indeed, looks like a FP problem with the sqrt()